src/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,109 @@
+import classNames from 'classnames';
+import {
+  FormEvent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../Types/Todo';
+
+/* eslint-disable jsx-a11y/control-has-associated-label */
+interface Props {
+  todo: Todo,
+  handleComplete: (selectedTodoId: number) => void,
+  handleDelete: (selectedTodoId: number) => void,
+  handleEdit: (selectedTodoId: number, editingValue: string) => void,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  handleComplete,
+  handleDelete,
+  handleEdit,
+}) => {
+  const { title, completed, id } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [editingValue, setEditingValue] = useState(title);
+  const editingInputRef = useRef<HTMLInputElement>(null);
+
+  const handleCompleteChange = (selectedTodoId: number) => {
+    handleComplete(selectedTodoId);
+  };
+
+  const handleDeleteClick = (selectedTodoId: number) => {
+    handleDelete(selectedTodoId);
+  };
+
+  const handleEditSumbit = (event: FormEvent, selectedTodoId: number) => {
+    event.preventDefault();
+
+    if (editingValue) {
+      handleEdit(selectedTodoId, editingValue);
+    } else {
+      handleDelete(selectedTodoId);
+    }
+
+    setIsEditing(false);
+  };
+
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setEditingValue(title);
+      setIsEditing(false);
+    }
+  };
+
+  const handleDoubleClick = () => {
+    setEditingValue(title);
+    setIsEditing(true);
+    setTimeout(() => {
+      editingInputRef.current?.focus();
+    }, 0);
+  };
```

Comment:
I've set it up locally and tested your solution and it works totally fine and meed, I wouldn't bother by the failing tests